Nadim.CinemaReservationSystem.Web/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Nadim.CinemaReservationSystem.Web.Models;
+using System.Security.Claims;
+using Microsoft.IdentityModel.Tokens;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.Configuration;
+using System.IdentityModel.Tokens.Jwt;
+using Microsoft.AspNetCore.Authorization;
+
+namespace Nadim.CinemaReservationSystem.Web.Controllers
+{
+    [Route("api/[controller]")]
+    public class AuthenticationController : Controller
+    {
+        private readonly CinemaReservationSystemContext dbContext;
+        public readonly IConfiguration configuration;
+        public AuthenticationController(CinemaReservationSystemContext context, IConfiguration configuration)
+        {
+            dbContext = context;
+            this.configuration = configuration;
+        }
+
+        [AllowAnonymous]
+        [HttpPost("[action]")]
+        public IActionResult Login([FromBody] User user)
+        {
+            if (!Regex.IsMatch(user.Email, @"^([\w-\.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([\w-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$") ||
```

Comment:
This method is quite long and doing too many different things, don't you  think?